<a href="https://colab.research.google.com/github/Aniket-Satpute/Airbnb-Booking-Analysis/blob/main/airbnb_booking_analysis_aniket.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **INTRODUCTION-**


 
## <b> Airbnb,is an American company that operates an online marketplace for lodging, primarily homestays for vacation rentals, and tourism activities.
## <b>Airbnb connects people who want to rent out their homes with people looking for accommodations in that locality. Newyork is the most populous city in the United States, and one of the most popular tourism and business places globally. 
## <b>Airbnb offers people an easy, relatively stress-free way to earn some income from their property.




# **SCOPE OF THE PROJECT-**

## <b>The objective of the project is obtain the meaningful analysis about the Airbnb Bookings with respect to the factors asscociated with it. Mainly focused on terms as below-**

1. Data Prepping/Wrangling.
2. Checking the Null values for cleaning the Dataset for further analysis
3. Checking the unique values for Analyzing the Dataset for further analysis
4. Exploration of Neighbourhood group,Neighbourhood,Roomtype,Price,Reviews.
5. Correlaton between variables.
6. Analysis of busy hosts and key findings.

# **PROBLEM STATEMENT-**

## <b> Since 2008, guests and hosts have used Airbnb to expand on traveling possibilities and present a more unique, personalized way of experiencing the world. Today, Airbnb became one of a kind service that is used and recognized by the whole world. Data analysis on millions of listings provided through Airbnb is a crucial factor for the company. These millions of listings generate a lot of data - data that can be analyzed and used for security, business decisions, understanding of customers' and providers' (hosts) behavior and performance on the platform, guiding marketing initiatives, implementation of innovative additional services and much more. </b>

## <b>This dataset has around 49,000 observations in it with 16 columns and it is a mix between categorical and numeric values. </b>

## <b> Explore and analyze the data to discover key understandings (not limited to these) such as : 
* What can we learn about different hosts and areas?
* What can we learn from predictions? (ex: locations, prices, reviews, etc)
* Which hosts are the busiest and why?
* Is there any noticeable difference of traffic among different areas and what could be the reason for it? </b>

  # **DATA PREPPING**

##**Mount A Drive-**

In [5]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


## **Importing the libraries-**

In [6]:
#Importing libraries for Exploratory data Analysis
import numpy as np # For some math operations on arrays in predictions.
import pandas as pd  # To upload the large data set and wrangling the data.
import matplotlib.pyplot as plt  # Used for visualization of dataset.
import seaborn as sns  # Used for visualization of dataset.

In [7]:
# Loading Data set of csv file
ABA = pd.read_csv("/content/drive/MyDrive/airbnb booking analysis github/Airbnb NYC 2019.csv")

## **Some Basic Information about Data set-**

In [8]:
ABA.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

In [9]:
# To understand some statistics of our dataset-
ABA.describe()

id       host_id      latitude     longitude         price  \
count  4.889500e+04  4.889500e+04  48895.000000  48895.000000  48895.000000   
mean   1.901714e+07  6.762001e+07     40.728949    -73.952170    152.720687   
std    1.098311e+07  7.861097e+07      0.054530      0.046157    240.154170   
min    2.539000e+03  2.438000e+03     40.499790    -74.244420      0.000000   
25%    9.471945e+06  7.822033e+06     40.690100    -73.983070     69.000000   
50%    1.967728e+07  3.079382e+07     40.723070    -73.955680    106.000000   
75%    2.915218e+07  1.074344e+08     40.763115    -73.936275    175.000000   
max    3.648724e+07  2.743213e+08     40.913060    -73.712990  10000.000000   

       minimum_nights  number_of_reviews  reviews_per_month  \
count    48895.000000       48895.000000       38843.000000   
mean         7.029962          23.274466           1.373221   
std         20.510550          44.550582           1.680442   
min          1.000000           0.000000           0.010000   
25%          1.000000           1.000000           0.190000   
50%          3.000000           5.000000           0.720000   
75%          5.000000          24.000000           2.020000   
max       1250.000000         629.000000          58.500000   

       calculated_host_listings_count  availability_365  
count                    48895.000000      48895.000000  
mean                         7.143982        112.781327  
std                         32.952519        131.622289  
min                          1.000000          0.000000  
25%                          1.000000          0.000000  
50%                          1.000000         45.000000  
75%                          2.000000        227.000000  
max                        327.000000        365.000000

# **Checking the Null values for cleaning the Dataset for further analysis.**

In [10]:
ABA.isnull()

id   name  host_id  host_name  neighbourhood_group  neighbourhood  \
0      False  False    False      False                False          False   
1      False  False    False      False                False          False   
2      False  False    False      False                False          False   
3      False  False    False      False                False          False   
4      False  False    False      False                False          False   
...      ...    ...      ...        ...                  ...            ...   
48890  False  False    False      False                False          False   
48891  False  False    False      False                False          False   
48892  False  False    False      False                False          False   
48893  False  False    False      False                False          False   
48894  False  False    False      False                False          False   

       latitude  longitude  room_type  price  minimum_nights  \
0         False      False      False  False           False   
1         False      False      False  False           False   
2         False      False      False  False           False   
3         False      False      False  False           False   
4         False      False      False  False           False   
...         ...        ...        ...    ...             ...   
48890     False      False      False  False           False   
48891     False      False      False  False           False   
48892     False      False      False  False           False   
48893     False      False      False  False           False   
48894     False      False      False  False           False   

       number_of_reviews  last_review  reviews_per_month  \
0                  False        False              False   
1                  False        False              False   
2                  False         True               True   
3                  False        False              False   
4                  False        False              False   
...                  ...          ...                ...   
48890              False         True               True   
48891              False         True               True   
48892              False         True               True   
48893              False         True               True   
48894              False         True               True   

       calculated_host_listings_count  availability_365  
0                               False             False  
1                               False             False  
2                               False             False  
3                               False             False  
4                               False             False  
...                               ...               ...  
48890                           False             False  
48891                           False             False  
48892                           False             False  
48893                           False             False  
48894                           False             False  

[48895 rows x 16 columns]

In [11]:
print(ABA.isnull().sum().sum())

20141


**Comments-We have 4 columns contain a missing value. 
chech number of missing value present in column = total is 20141
from 4 column 2,4,12,13**

In [12]:
ABA.isnull().sum()

id                                    0
name                                 16
host_id                               0
host_name                            21
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       10052
reviews_per_month                 10052
calculated_host_listings_count        0
availability_365                      0
dtype: int64

**Now, to remove the Null values- we replace the null values**

In [13]:
ABA['name'].fillna('not_known',inplace=True)
ABA['host_name'].fillna('not_known',inplace=True)
ABA['last_review'].fillna('unknown',inplace=True)
ABA['reviews_per_month'].fillna('unknown',inplace=True)

In [14]:
# Again checking the Null values after replacing it in Dataset-
ABA.isnull().sum()

id                                0
name                              0
host_id                           0
host_name                         0
neighbourhood_group               0
neighbourhood                     0
latitude                          0
longitude                         0
room_type                         0
price                             0
minimum_nights                    0
number_of_reviews                 0
last_review                       0
reviews_per_month                 0
calculated_host_listings_count    0
availability_365                  0
dtype: int64